Fhi.HelseId/Api/Authorization/CustomAuthenticationHandler.cs

```diff
@@ -0,0 +1,203 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using Microsoft.IdentityModel.Tokens;
+
+
+namespace Fhi.HelseId.Api.Authorization
+{
+    /// <summary>
+    /// This is taken from package: Microsoft.AspNetCore.Authentication.JwtBearer, class JwtBearerHandler.
+    /// See the code by going into the definition, or this link: https://github.com/dotnet/aspnetcore/blob/99f2ba5fd1d668fecbff1e1a9ba6a541c99bec79/src/Security/Authentication/JwtBearer/src/JwtBearerHandler.cs#L15
+    /// !IMPORTANT! the only thing we to is to add DPoP for authorazation
+    /// </summary>
+    public class CustomAuthenticationHandler(
+        IOptionsMonitor<JwtBearerOptions> options,
+        ILoggerFactory logger,
+        UrlEncoder encoder,
+        ISystemClock clock) : JwtBearerHandler(options, logger, encoder, clock)
+    {
+        private OpenIdConnectConfiguration? _configuration;
+
+        /// <summary>
+        /// The only change is to add authorization.StartsWith("DPoP ")
+        /// </summary>
+        /// <returns></returns>
+        protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
+        {
+            string? token = null;
+            try
+            {
+                // Give application opportunity to find from a different location, adjust, or reject token
+                var messageReceivedContext = new MessageReceivedContext(Context, Scheme, Options);
+
+                // event can set the token
+                await Events.MessageReceived(messageReceivedContext);
+                if (messageReceivedContext.Result != null)
+                {
+                    return messageReceivedContext.Result;
+                }
+
+                // If application retrieved token from somewhere else, use that.
+                token = messageReceivedContext.Token;
+
+                if (string.IsNullOrEmpty(token))
+                {
+                    string authorization = Request.Headers.Authorization;
+
+                    // If no authorization header found, nothing to process further
+                    if (string.IsNullOrEmpty(authorization))
+                    {
+                        return AuthenticateResult.NoResult();
+                    }
+
+                    if (authorization.StartsWith("Bearer ", StringComparison.OrdinalIgnoreCase))
+                    {
+                        token = authorization.Substring("Bearer ".Length).Trim();
+                    }
+
+                    // THIS IS THE ONLY CHANGE
+                    if (authorization.StartsWith("DPoP ", StringComparison.OrdinalIgnoreCase))
```

Comment:
Vær forsiktige med dette. Her har dere åpnet for at klienten kan kalle endepunktet deres med et DPoP token, men hvis dere skal tillate dette så må dere også validere at DPoP-tokenet er gyldig. 
Hvis hensikten er å la endepunktet goda dpop-tokens på samme måte som bearer-tokens, så er løsningen å sende dpop-tokenet med Bearer-prefixet.